src/main/game/entities/units/Unit.java

```diff
@@ -96,8 +100,29 @@ else if(location.equals(new Location(this.location.getX()-1, this.location.getY(
     /* iMoveable */
     public int getMoveDistance(){ return stats.getSpeed(); }
 
+    /* Stat-adjusted damage taking */
+    @Override
+    public void takeDamage(double damage){
```

Comment:
can you just put this in entity instead of overriding since you do the same thing for both unit and structure?